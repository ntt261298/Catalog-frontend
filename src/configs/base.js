export default {
  apiHost: 'http://localhost:5000',
};
