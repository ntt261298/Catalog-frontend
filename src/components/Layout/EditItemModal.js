import React from 'react';
import { connect } from 'react-redux';
import {
  Button, Modal, ModalHeader, ModalBody, ModalFooter,
} from 'reactstrap';
import { PropTypes } from 'prop-types';
import { hideModal } from 'actions/app';
import getCategories from 'actions/category';
import { editItem, getItems, getUserItems } from 'actions/item';
import { selectAllCategories } from 'utils/selector';
import { errMessage, successMessage } from 'utils/messages';


export class EditItemModal extends React.Component {
    state = {
      title: '',
      description: '',
      categoryID: '',
      itemID: '',
    }

    componentDidMount() {
      this.props.getCategories();
      this.setState(this.props.item);
    }

    changeInput = (e) => {
      this.setState({
        [e.target.name]: e.target.value,
      });
    }

    editItem = async () => {
      const {
        title, description, categoryID, itemID,
      } = this.state;
      // Set Item for comparing with current Item from props
      const item = {
        itemID,
        title,
        description,
        categoryID,
      };
      if (title.trim() === '') {
        errMessage('Title must not be blank');
        return;
      }
      if (description.trim() === '') {
        errMessage('Description must not be blank');
        return;
      }

      await this.props.editItem(item)
        .then(message => successMessage(message))
        .catch(err => errMessage(err));
      await this.props.getUserItems()
        .catch(err => errMessage(err));
      this.props.hideModal();
    }

    render() {
      const { modal, hideModal, categories } = this.props;
      const {
        title, description, categoryID,
      } = this.state;
      const isEditItemModal = modal === 'editItem';
      return (
        <div>
          <Modal isOpen={isEditItemModal} toggle={hideModal}>
            <ModalHeader toggle={hideModal}>Edit Item</ModalHeader>
            <ModalBody>
              <div>
                <label htmlFor="title">
                    Title:
                  <input type="text" name="title" value={title} onChange={this.changeInput} />
                </label>
              </div>
              <div>
                <label htmlFor="description">
                    Description:
                  <br />
                  <textarea name="description" value={description} onChange={this.changeInput} />
                </label>
              </div>
              <div>
                <label htmlFor="categoryID">
                    Category:
                  <select name="categoryID" value={categoryID} onChange={this.changeInput}>
                    {
                        categories.map(
                          ({ id, name }) => <option key={id} value={id}>{name}</option>,
                        )
                    }
                  </select>
                </label>
              </div>
            </ModalBody>
            <ModalFooter>
              <Button color="primary" onClick={this.editItem}>Edit</Button>
            </ModalFooter>
          </Modal>
        </div>
      );
    }
}

EditItemModal.propTypes = {
  modal: PropTypes.string.isRequired,
  item: PropTypes.object.isRequired,
  categories: PropTypes.array.isRequired,
  hideModal: PropTypes.func.isRequired,
  editItem: PropTypes.func.isRequired,
  getCategories: PropTypes.func.isRequired,
  getUserItems: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  modal: state.app.modal,
  item: state.user.currentItem,
  categories: selectAllCategories(state),
});

export default connect(mapStateToProps, {
  getCategories, getItems, getUserItems, editItem, hideModal,
})(EditItemModal);
