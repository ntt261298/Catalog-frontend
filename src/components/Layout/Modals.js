import React from 'react';
import Loadable from 'react-loadable';


export const LoginModal = Loadable({
  loader: () => import('components/Layout/LoginModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
export const SignupModal = Loadable({
  loader: () => import('components/Layout/SignupModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
export const AddItemModal = Loadable({
  loader: () => import('components/Layout/AddItemModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
export const EditItemModal = Loadable({
  loader: () => import('components/Layout/EditItemModal'),
  loading: () => <div>Loading...</div>,
  delay: 200,
});
