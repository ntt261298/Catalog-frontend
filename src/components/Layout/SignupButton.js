import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { showSignup } from '../../actions/app';

export const SignupButton = props => (
  <button type="button" onClick={props.showSignup} className="btn-signup">Signup</button>
);

SignupButton.propTypes = {
  showSignup: PropTypes.func.isRequired,
};

export default connect(null, { showSignup })(SignupButton);
