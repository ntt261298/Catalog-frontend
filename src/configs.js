import {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Enzyme config
configure({ adapter: new Adapter() });

// API host config
const defaultHost = 'http://localhost:5000';

export {
  shallow, mount, render, defaultHost,
};
