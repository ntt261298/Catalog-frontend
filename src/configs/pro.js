import base from 'configs/base';

export default {
  ...base,
};
