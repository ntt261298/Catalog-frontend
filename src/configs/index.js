import {
  configure, shallow, mount, render,
} from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import dev from 'configs/dev';
import pro from 'configs/pro';

// Enzyme config
configure({ adapter: new Adapter() });

let configs;

if (process.env.REACT_APP_ENV === 'dev') {
  configs = dev;
} else if (process.env.REACT_APP_ENV === 'pro') {
  console.log('pro');
  configs = pro;
} else {
  // Default configs is dev
  configs = dev;
}

export { shallow, mount, render };

export default configs;
