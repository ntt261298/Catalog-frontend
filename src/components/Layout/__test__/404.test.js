import React from 'react';
import { shallow } from 'configs';
import { Page404 } from 'components/Layout/404';

describe('components/Layout/404', () => {
  let wrapper;

  const setup = () => {
    wrapper = shallow(
      <Page404 />,
    );
  };

  it('should render 404 page', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
