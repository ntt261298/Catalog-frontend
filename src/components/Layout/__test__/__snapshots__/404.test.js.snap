// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout/404 should render 404 page 1`] = `
<div
  className="stage"
>
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
  <div
    className="layer"
  />
</div>
`;
